tests/category-endpoints/category.test.ts

```diff
@@ -0,0 +1,605 @@
+            desc += desc
+          }
+          await runTest({
+            ...baseResponse,
+            pictureUrl: desc
+          }, {
+            field: 'pictureUrl',
+            validationCode: 'pictureUrl.LENGTH_NOT_VALID'
+          })
+        })
+
+        it('verifies that the string is an URL', async () => {
+          await runTest({
+            ...baseResponse,
+            pictureUrl: 'testy'
+          }, {
+            field: 'pictureUrl',
+            validationCode: 'pictureUrl.URL'
+          })
+        })
+      })
+
+      describe('Returns 400 when `color` is invalid', () => {
+        const baseResponse = {
+          name: 'Sample name',
+          type: 'TO_USER',
+          description: 'Sample description',
+          pictureUrl: 'http://google.com',
+          extra: 'Extra information',
+        }
+
+        it('verifies that the field is on the request', async () => {
+          await runTest(baseResponse, {
+            field: 'color',
+            validationCode: 'color.REQUIRED'
+          })
+        })
+
+        it('verifies that the field is a string', async () => {
+          await runTest({
+            ...baseResponse,
+            color: 1
+          }, {
+            field: 'color',
+            validationCode: 'color.STRING'
+          })
+        })
+
+        it('verifies that the string doesn\'t exceed 50 chars', async () => {
+          await runTest({
+            ...baseResponse,
+            color: '1234567812345678123456781234567812345678123456781234567812345678'
+          }, {
+            field: 'color',
+            validationCode: 'color.LENGTH_NOT_VALID'
+          })
+        })
+
+        it('verifies that the string is not empty', async () => {
+          await runTest({
+            ...baseResponse,
+            color: ''
+          }, {
+            field: 'color',
+            validationCode: 'color.LENGTH_NOT_VALID'
+          })
+        })
+      })
+
+      describe('Returns 400 when `extra` is invalid', () => {
+        const baseResponse = {
+          type: 'TO_USER',
+          pictureUrl: 'http://google.com',
+          description: 'Sample description',
+          name: 'Test name',
+          color: '#000000'
+        }
+
+        it('verifies that the field is a string', async () => {
+          await runTest({
+            ...baseResponse,
+            extra: 1
+          }, {
+            field: 'extra',
+            validationCode: 'extra.STRING'
+          })
+        })
+
+        it('verifies that the string doesn\'t exceed 255 chars', async () => {
+          let extra = 'abcde'
+          while (extra.length < 256) {
+            extra += extra
+          }
+
+          await runTest({
+            ...baseResponse,
+            extra: extra
+          }, {
+            field: 'extra',
+            validationCode: 'extra.LENGTH_NOT_VALID'
+          })
+        })
+      })
+    })
+  })
+
+  describe('Category lookup by ID', () => {
```

Comment:
On this endpoint we need tests for non-number lookup (it would return 404)

Reply:
Sovled with  b64ae39

Reply:
Note however that the test returns a 400 error, given that the error is that the url is misformed (should have an INT according to the validator) and therefore it's a bad request